test/store_test.dart

```diff
@@ -33,7 +33,7 @@ void main() {
     });
 
     test('should trigger with itself as the payload', () async {
-      Completer c = new Completer();
+      var c = new Completer();
       store.listen((Store payload) {
```

Comment:
Since you are testing that the anonymous function executes, would it make sense to include the call to `expectAsync()` to make sure that things inside `expectAsync()` are being called as expected?